crates/hir-ty/src/display.rs

```diff
@@ -1642,101 +1760,861 @@ impl HirDisplay for Ty {
     }
 }
 
-fn hir_fmt_generics(
-    f: &mut HirFormatter<'_>,
-    parameters: &[GenericArg],
-    generic_def: Option<hir_def::GenericDefId>,
-    self_: Option<&Ty>,
-) -> Result<(), HirDisplayError> {
-    if parameters.is_empty() {
-        return Ok(());
-    }
-
-    let parameters_to_write = generic_args_sans_defaults(f, generic_def, parameters);
-
-    if !parameters_to_write.is_empty() {
-        write!(f, "<")?;
-        hir_fmt_generic_arguments(f, parameters_to_write, self_)?;
-        write!(f, ">")?;
-    }
+impl<'db> HirDisplay for crate::next_solver::Ty<'db> {
```

Comment:
I think you should replace the chalk ty display impl to converting to new solver ty then displaying it, to reduce duplicate code and maintenance efforts.

Reply:
It will also help us better test the code.

Reply:
I forwarded these impls. Associated type printing changes (only in tests). Had to add some `salsa::attach` calls, but otherwise no problems.